### PR TITLE
feat: Add user callback for handling invalid characters.

### PR DESCRIPTION
**Pull request by yosion-p**
hi buddy,this is work-in progress.
It might not be easy to get the user to write a callback function, but I think we can at least do it first.

I have noticed that there are many issues about callback(#53)  at present,
It may be difficult to solve all the problems, but we can take them apart step by step,
So now I have taken the first step (at least we can solve #210  for now),
and the first step now seems easy(or too easy🤣).

First, we declare and tell the user what custom rules can be processed:
```js
iconv.invalidChar;
iconv.ignore = '';
// ...other rules
// Sorry, I don't know how to describe it here, maybe TS would be better
```
Then, the options argument can be passed in when the user calls public API, like:
```js
iconv.decode(Buffer, encoding,{
  invalidChar: function(){/**/},
  ignore: someRules,
  // ...other rules
})
```
Finally, we pick it up at decode:
```js
iconv.decode = function decode(buf, encoding, options) {
// ...
if(options && options.invalidChar)  // ...
// ...
}
```
and the encoding is written similarly.

In short, we accept some special rules that user-written callback functions.

### CODE REVIEW

**Comment by yosion-p** [comment]
HI,bro @ashtuchkin 
Do you mind if I try to do that?

**Comment by ashtuchkin** [comment]
Hey, I think it's a good intention, but it's also a pretty complex topic (that's why I didn't implement it back in the days) and I don't have enough time these days to actively review or discuss design. Maybe in a couple months when the load on my day job decreases..

I don't think your current approach would work because '�' character can be in the input and we don't want to call the callback in that case.

**Comment by ashtuchkin** [comment]
>  i'd love to contribute, no matter it's easy or hard.
I understand that, and I think it's a great intention and a great way to learn. I appreciate that you're doing this. The problem is that it would require time commitment from my side to take this PR to a level at which I would feel comfortable merging it. This issue is more complex than your previous pull request. I don't have a good design in my head that would resolve it in a good way, so to even start code reviewing I would need to spend time thinking about it / designing it. I don't think an easy solution exists here that would not harm baseline performance and would avoid adding unexpected behavior.

> May I just throw an error instead of calling the callback when '�' character be inputed?
No, that would be unexpected behavior for the library users and would potentially break existing use cases.
